### PR TITLE
[opencascade] Fix opencascade debug library linkage

### LINKED ISSUE #16648: [opencascade] Linking against wrong freetype build type
Opencascade is currently linking agaisnt "freetype.dll" instead of "freetyped.dll" for debug libraries.

image

Just leaving the footnote here, I will probably try and submit a PR for it later

### PR DESCRIPTION
**Pull request by marekr**
Fixes #16648

opencascade was linking debug builds agaisnt release freeimage, freetype and tbb libraries.

### CODE REVIEW

File: ports/opencascade/fix-debug-library-find.patch [hunk 1]
```diff
@@ -0,0 +1,100 @@
+diff --git a/adm/cmake/3rdparty_macro.cmake b/adm/cmake/3rdparty_macro.cmake
+index a3621f762..1bdb8aba7 100644
+--- a/adm/cmake/3rdparty_macro.cmake
++++ b/adm/cmake/3rdparty_macro.cmake
+@@ -120,7 +120,7 @@ macro (THIRDPARTY_PRODUCT PRODUCT_NAME HEADER_NAME LIBRARY_CSF_NAME LIBRARY_NAME
+ #    if (BUILD_SHARED_LIBS)
+       # library
+       if (NOT 3RDPARTY_${PRODUCT_NAME}_LIBRARY_${LIBRARY_NAME_SUFFIX} OR NOT EXISTS "${3RDPARTY_${PRODUCT_NAME}_LIBRARY_${LIBRARY_NAME_SUFFIX}}")
+-        set (CMAKE_FIND_LIBRARY_SUFFIXES .lib .so .dylib .a)
++        set (CMAKE_FIND_LIBRARY_SUFFIXES ${LIBRARY_NAME_DEBUG_SUFFIX}.lib .lib .so .dylib .a)
```
Comment (NancyLi1013) [inline ports/opencascade/fix-debug-library-find.patch:10]: Do you know if this macro `LIBRARY_NAME_DEBUG_SUFFIX` will add the suffix '-d' for all dependent ports?

File: ports/opencascade/fix-debug-library-find.patch [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+   #ifdef _MSC_VER
+-    #pragma comment( lib, "FreeImage.lib" )
++    #ifdef _DEBUG
++        #pragma comment( lib, "FreeImaged.lib" )
++    #else
++        #pragma comment( lib, "FreeImage.lib" )
++    #endif
+   #endif
+ #elif defined(HAVE_WINCODEC)
+   #include <wincodec.h>
```
Comment (NancyLi1013) [inline ports/opencascade/fix-debug-library-find.patch:100]: Can you submit these changes to upstream?

File: ports/opencascade/CONTROL [hunk 3]
```diff
@@ -1,5 +1,6 @@
 Source: opencascade
 Version: 7.5.0
+Port-Version: 1
 Build-Depends: freetype
 Description: Open CASCADE Technology (OCCT) is an open-source software development platform for 3D CAD, CAM, CAE.
```
Comment (NancyLi1013) [inline ports/opencascade/CONTROL:5]:
```suggestion
Description: Open CASCADE Technology (OCCT) is an open-source software development platform for 3D CAD, CAM, CAE.
Homepage: https://github.com/Open-Cascade-SAS/OCCT
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Hi @marekr 

Thanks for your PR to fix this problem.

For the failure on x86-windows:

```
Error: While reading versions for port opencascade from file: C:\a\1\s\versions\o-\opencascade.json
       Version `7.5.0#1` was not found in versions file.
       Run:

           vcpkg x-add-version opencascade

       to add the new port version.
```
You can run `vcpkg x-add-version opencascade
` and then submit the changes for version.

Note: Please run the command after you updated all your changes.

**Comment by NancyLi1013** [comment]
@marekr Is work still being done for this PR?

File: ports/opencascade/fix-debug-library-find.patch (hunk 2 above)
Comment (marekr) [reply ports/opencascade/fix-debug-library-find.patch:100]: That is going to be quite an adventure, they don't accept pull requests and instead have a very...bureaucratic solution

File: ports/opencascade/fix-debug-library-find.patch (hunk 1 above)
Comment (marekr) [reply ports/opencascade/fix-debug-library-find.patch:10]: What do you mean?

**Comment by NancyLi1013** [comment]
Can you merge master to resolve the conflicts?

File: ports/opencascade/fix-debug-library-find.patch (hunk 1 above)
Comment (NancyLi1013) [reply ports/opencascade/fix-debug-library-find.patch:10]: Sorry for the long delay. I mean whether all thirdParty ports will be added the suffix '-d' in debug type.

**Comment by NancyLi1013** [comment]
Sorry again, there are new conflicts that need to be resolved. @marekr

**Comment by JonLiu1993** [comment]
@marekr , Is this PR being worked on? Could you please resolve the conflicts against master?

**Comment by Cheney-W** [comment]
Conveted this PR to draft since it seems that no progress is being made.

**Comment by Cheney-W** [comment]
Pinging @marekr  for response. Is work still being done for this PR?

**Comment by Cheney-W** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**